src/server/game/Maps/AreaBoundary.h

```diff
@@ -40,7 +41,7 @@ class AreaBoundary
             double d_positionX, d_positionY, d_positionZ;
             DoublePosition(double x = 0, double y = 0, double z = 0, float o = 0)
                 : Position((float)x, (float)y, (float)z, o), d_positionX(x), d_positionY(y), d_positionZ(z) { }
-            DoublePosition(float x = 0, float y = 0, float z = 0, float o = 0)
+            DoublePosition(float x, float y = 0, float z = 0, float o = 0)
```

Comment:
is this some kind of trick to distinguish between double and float overloads when no arguments are passed ?